src/collection.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+
+import { v4 as uuid4 } from "uuid";
+import { attachFakeIDBSymbolsTo } from "./utils";
+
+attachFakeIDBSymbolsTo(typeof global === "object" ? global : window);
+
+export default class Collection {
+  constructor(collName) {
+    this._collName = collName;
+    this._db;
+  }
+
+  get name() {
+    return this._collName;
+  }
+
+  init() {
+    return new Promise((resolve, reject) => {
+      var request = indexedDB.open(this._collName, 1);
+      request.onupgradeneeded = event => {
+        var store = event.target.result.createObjectStore(this._collName, {
+          keyPath: "id"
+        });
+        store.createIndex("id", "id", { unique: true });
+      };
+      request.onerror = event => {
+        reject(event.error);
+      };
+      request.onsuccess = event => {
+        this._db = event.target.result;
+        resolve(this);
+      };
+    });
+  }
+
+  transaction(mode) {
+    return this._db
+      .transaction([this._collName], mode)
+      .objectStore(this._collName);
```

Comment:
This method is called transaction but actually starts a new transaction and returns an object store.